bindings/go/scip/symbol.go

```diff
@@ -21,253 +20,55 @@ func IsLocalSymbol(symbol string) bool {
-}
-
-func (s *symbolParser) current() rune {
-	if s.index < len(s.Symbol) {
-		return s.Symbol[s.index]
-	}
-	return '\x00'
-}
-
-func (s *symbolParser) peekNext() rune {
-	if s.index+1 < len(s.Symbol) {
-		return s.Symbol[s.index]
-	}
-	return 0
-}
-
-func (s *symbolParser) parseDescriptors() ([]*Descriptor, error) {
-	var result []*Descriptor
-	for s.index < len(s.Symbol) {
-		descriptor, err := s.parseDescriptor()
-		if err != nil {
-			return nil, err
-		}
-		result = append(result, descriptor)
-	}
-	return result, nil
-}
-
-func (s *symbolParser) parseDescriptor() (*Descriptor, error) {
-	start := s.index
-	switch s.peekNext() {
-	case '(':
-		s.index++
-		name, err := s.acceptIdentifier("parameter name")
-		if err != nil {
-			return nil, err
-		}
-		return &Descriptor{Name: name, Suffix: Descriptor_Parameter}, s.acceptCharacter(')', "closing parameter name")
-	case '[':
-		s.index++
-		name, err := s.acceptIdentifier("type parameter name")
-		if err != nil {
-			return nil, err
-		}
-		return &Descriptor{Name: name, Suffix: Descriptor_TypeParameter}, s.acceptCharacter(']', "closing type parameter name")
-	default:
-		name, err := s.acceptIdentifier("descriptor name")
-		if err != nil {
-			return nil, err
-		}
-		suffix := s.current()
-		s.index++
-		switch suffix {
-		case '(':
-			disambiguator := ""
-			if s.peekNext() != ')' {
-				disambiguator, err = s.acceptIdentifier("method disambiguator")
-				if err != nil {
-					return nil, err
-				}
-			}
-			err = s.acceptCharacter(')', "closing method")
-			if err != nil {
-				return nil, err
-			}
-			return &Descriptor{Name: name, Disambiguator: disambiguator, Suffix: Descriptor_Method}, s.acceptCharacter('.', "closing method")
-		case '/':
-			return &Descriptor{Name: name, Suffix: Descriptor_Namespace}, nil
-		case '.':
-			return &Descriptor{Name: name, Suffix: Descriptor_Term}, nil
-		case '#':
-			return &Descriptor{Name: name, Suffix: Descriptor_Type}, nil
-		case ':':
-			return &Descriptor{Name: name, Suffix: Descriptor_Meta}, nil
-		case '!':
-			return &Descriptor{Name: name, Suffix: Descriptor_Macro}, nil
-		default:
-		}
-	}
-
-	end := s.index
-	if s.index > len(s.Symbol) {
-		end = len(s.Symbol)
-	}
-	return nil, errors.Newf("unrecognized descriptor %q", string(s.Symbol[start:end]))
-}
-
-func (s *symbolParser) acceptIdentifier(what string) (string, error) {
-	if s.current() == '`' {
-		s.index++
-		return s.acceptBacktickEscapedIdentifier(what)
-	}
-	start := s.index
-	for s.index < len(s.Symbol) && isSimpleIdentifierCharacter(s.current()) {
-		s.index++
-	}
-	if start == s.index {
-		return "", s.error("empty identifier")
-	}
-	return string(s.Symbol[start:s.index]), nil
+// ParseSymbolUTF8 parses an SCIP string into the Symbol message.
+//
+// Unlike ParseSymbol, this skips UTF-8 validation. To customize
+// parsing behavior, use ParseSymbolUTF8With.
+func ParseSymbolUTF8(symbol beaut.UTF8String) (*Symbol, error) {
```

Comment:
I've decided to add a new function here instead of modifying the signature of the existing `ParseSymbol` function to avoid gratuitously breaking callers, since it's not hard to maintain back-compat.